enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,517 @@
+
+- Deal with users who want their ingress traffic to not go through the single
+control-plane node. Such users can easily achieve that goal by modifying the
+`IngressController` to target just the worker, non-control-plane node(s)
+(perhaps by labeling all of them with a unique label that is not applied to the
+control-plane node that is also a worker and then have the `IngressController`
+target that label).
+
+- Deal with non-cloud, non-"none" platforms such as baremetal, vSphere, etc.
+
+## Proposal
+
+- Create a new `DefaultPlacement` API field.
+
+- Make sure worker node ignition files are generated even in bootstrap-in-place
+single control-plane node "none"-platform installation.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to install a single control-plane
+node cluster with additional worker nodes, so that my cluster will be able to
+handle larger computation demands.
+
+- As an OpenShift cluster administrator, I want to add worker nodes to my
+existing single control-plane node cluster, so that it'll be able to meet
+growing computation demands.
+
+### API Extensions
+
+Introduce a new status field in the Ingress config CR
+(`config.openshift.io/v1/ingresses`) called `DefaultPlacement`.
+
+In addition, continue to allow the `.spec.replicas` and `.spec.nodePlacement`
+fields in `operator.openshift.io/v1/ingresscontrollers` CRs to be omitted, but
+change the defaulting behavior for these fields based on the new status field.
+
+The sections below go into detail about the field, its possible values, and the
+behavior expected from each of them, but in practice following is the proposed
+addition to `openshift/api`'s `config/v1/types_ingress.go` file:
+
+```go
+type IngressStatus struct {
+    // ... existing fields omitted
+
+	// defaultPlacement is set by the installer and lets it control which
+	// nodes will host the ingress router pods by default, the options being
+	// control-plane nodes or worker nodes.
+	//
+	// This field works by dictating how the Ingress Operator will set the default
+	// values of future IngressController resources' replicas and nodePlacement
+	// fields.
+	//
+	// The value of replicas is set based on the value of a chosen field in the
+	// Infrastructure CR. If defaultPlacement was set to ControlPlane, the
+	// chosen field will be controlPlaneTopology. If it was set to Workers the
+	// chosen field will be infrastructureTopology. Replicas will then be set to 1
+	// or 2 with respect to the chosen field's value (which can be either
+	// SingleReplica or HighlyAvailable).
+	//
+	// The value of nodePlacement is adjusted based on defaultPlacement - If
+	// defaultPlacement is set to ControlPlane the "node-role.kubernetes.io/worker"
+	// label will be added. If defaultPlacement is set to Workers the
+	// "node-role.kubernetes.io/master" label will be added.
+	//
+	// +kubebuilder:validation:Enum:="ControlPlane";"Workers"
+	// +kubebuilder:default:="Workers"
```

Comment:
The default value isn't included in auto generated docs, so you may want to add something like below to the godoc string so that the API reference tells users what the default is
```
// When omitted, the default value is Workers.
```